File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowel = {'a','e','i','o','u'}

        n = len(s)

        i=0
        j=0

        cnt = 0
        max_vowel = 0

        while j<n:
            if s[j] in vowel:
                cnt +=1
            if j-i+1 == k:
                max_vowel = max(max_vowel,cnt)

                if s[i] in vowel:
                    cnt-=1
                i+=1
            j+=1
        return max_vowel
        

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    bool isvowel(char ch){
        return ch=='a'|| ch=='e'|| ch=='i'|| ch=='o'|| ch=='u';
    }
    int maxVowels(string s, int k) {
        int n= s.size();
        int i=0;
        int j=0;

        int sum = 0;
        int maxv = 0;

        while(j<n){
            if(isvowel(s[j]))sum++;
            if(j-i+1==k){
                maxv = max(maxv,sum);

                if(isvowel(s[i]))sum--;
                i++;
            }
            j++;
        }
        return maxv;
        
    }
};